internal/operator/client.go

```diff
@@ -79,29 +76,54 @@ func NewOpCapClient() (Client, error) {
 	return operatorClient, nil
 }
 
-func NewOlmClientset() (*olmclient.Clientset, error) {
-	cfg, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+// kubeConfig return kubernetes cluster config
+func kubeConfig() (*rest.Config, error) {
+	return ctrl.GetConfig()
+}
+
+// mustGetConfig returns only the kubernetes cluster config.
+// nil is returned when an error is encountered
+func mustGetConfig() *rest.Config {
+	config, err := kubeConfig()
 	if err != nil {
-		return nil, err
+		return nil
 	}
+	return config
+}
```

Comment:
Hi @OchiengEd , if it's not being used it shouldn't be there. Let's try to get the code less confusing as possible. We already need to refactor a lot of things and this would add more complexity with no need. The errors should be handled always and if a client didn't get build we should have a nice message to the user and/or developer on debug mode according to the case.